public/app/features/alerting/unified/mocks/server/handlers/alertmanagers.ts

```diff
@@ -58,11 +62,29 @@ export const alertmanagerAlertsListHandler = () =>
     ]);
   });
 
-export const getGrafanaAlertmanagerConfigHandler = (config: AlertManagerCortexConfig = alertmanagerConfigMock) =>
-  http.get('/api/alertmanager/grafana/config/api/v1/alerts', () => HttpResponse.json(config));
+export const getAlertmanagerConfigHandler = () =>
+  http.get<{ name: string }>('/api/alertmanager/:name/config/api/v1/alerts', ({ params }) => {
```

Comment:
I removed the separate handlers for the `grafana` case, and instead consolidated to one endpoint that can just branch within to handle grafana/other alertmanagers. This is cleaner and we don't run as much of a risk of accidentally having handlers in the wrong order (if you put the `.../:name` case before `.../grafana`, then the logic in the first would take precedence and your other handler would be unused)